infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -141,280 +158,315 @@ private boolean isBootstrapped(CorfuMsg msg, ChannelHandlerContext ctx, IServerR
         return true;
     }
 
+    private boolean isBootstrapped(HeaderMsg reqHeader, ChannelHandlerContext ctx, IServerRouter r) {
```

Comment:
Instead of calling this from each handler, you can just move it to `processRequest`. Note that it needs to be called on all messages except for `LAYOUT_NOBOOTSTRAP`

Reply:
Note that checking this in `processRequest `means we are performing the check before the task is submitted to the executor service.  Is it possible that the status changes by the time the new thread is actually executing the handler?